src/v/cluster/archival_metadata_stm.cc

```diff
@@ -204,6 +204,14 @@ command_batch_builder& command_batch_builder::reset_metadata() {
     return *this;
 }
 
+command_batch_builder& command_batch_builder::update_start_kafka_offset(kafka::offset o) {
```

Comment:
should there be a corresponding place in the stm that acts upon detecting this type of command?